conans/test/unittests/model/transitive_reqs_test.py

```diff
@@ -1498,7 +1504,10 @@ class LibDConan(ConanFile):
         libd_ref = ConanFileReference.loads("LibD/0.1@user/testing")
         self.retriever.save_recipe(libd_ref, libd_content)
 
-        with six.assertRaisesRegex(self, ConanException, "Conflict in LibB/0.1@user/testing"):
+        with six.assertRaisesRegex(self, ConanException, "Conflict in LibB/0.1@user/testing\n"
```

Comment:
I am not sure I get this message. It says a conflict in LibB/0.1 against a requirement defined by itself?

Reply:
I still don't understand this.

``LibA/0.2@user/testing -> LibA/0.2@user/testing``?
Is there a cyclic dependency LibA->LibB->LibA???

Reply:
No, it should a root node actually.

LibA/0.2@user/testing -> root
LibA/0.2@user/testing -> LibB/0.1@user/testing
LibB/0.1@user/testing -> LibA/0.1@user/testing

The idea is illustrate that  LibA/0.2@user/testing in under conflict with LibA/0.1@user/testing due LibB/0.1@user/testing

Reply:
I raise the stakes and ask if this is a conflict :see_no_evil: 

If I'm not missing anything, this is the base graph of dependencies before this test:

![image](https://user-images.githubusercontent.com/1406456/75696728-56e1f280-5cac-11ea-9199-77c8d0c78c02.png)

This test modifies `LibD` with `requires = "LibB/0.1@user/testing", ("LibA/0.2@user/testing", "override")` so:

![image](https://user-images.githubusercontent.com/1406456/75697012-c821a580-5cac-11ea-9f39-3d504509da01.png)

We can consider two scenarios:
 * if `LibD` is overriding requirements from `LibB` there is no such a conflict, as both should point to `LibA/0.2` now
 * every path to root (to `consumer`) defines a different set of dependencies, then we have two paths and one of them doesn't go through `LibD` so the dependency to `LibA` is not overridden in that one. In that case this message is ok, as the conflict is between `LibB` requiring `LibA/0.1` (`LibB` is not overridden) and `LibB` requiring `LibA/0.2` (if the library is overridden).

Please, tell me that my drawing is wrong, otherwise I would think that here we are not resolving the graph properly.

Reply:
Yes, this is still a conflict. When there are diamond structures, the overrides, or evaluated transitive dependencies should be the same. In the chain consumer->LibC->LibB, LibC is expecting to use LibA/0.1, there is no one in the chain that  specifies otherwise. A newly introduced branch LibD by the consumer shouldn't be changing that behavior without the consumer knowing. An example:
- You have your app, that depends on Boost/1.70, which depends on OpenSSL/1.1 which depends on Zlib/1.2.11. All works good.
- Now you introduce in your app a new dependency to Poco/1.2 that depends on OpenSSL/1.1. but also has an override to ZLib/1.2.8. You want to detect that, not to downgrade to zlib/1.2.8 without noticing.

Summary: parallel branches in the dependency graph should agree on the versions or they will conflict, they cannot override each other, as there is no "priority" among them.

Reply:
Ok. Then it is a conflict and it is indeed a conflict between `LibB` and `LibB` (it would be really nice to improve the message, but it will require navigating through the graph and right now it is quite difficult).